File: problems/1/haskell/Multiples.hs
module Multiples
( sumMultiples
) where

sumMultiples :: [Int] -> Int -> Int
sumMultiples multiples ceiling =
    foldl (+) 0 $ [ x | x <- [0..ceiling], divisable multiples x ]

divisable :: [Int] -> Int -> Bool
divisable multiples num  =
    any divisable' multiples
    where divisable' = \x -> num `mod` x == 0

File: problems/1/javascript/multiples.js
const sumMultiples = (multiples, ceiling) => {
    return Array.apply(null, {length: ceiling})
        .map(Number.call, Number)
        .filter((num) => isMultipleOf(num, multiples))
        .reduce((sum, num) => sum + num, 0);
}

const isMultipleOf = (num, multiples) => {
    return multiples.some((multiple) => num % multiple === 0);
}

module.exports = sumMultiples;

console.log(sumMultiples([3,5], 1000));